### DISCUSSION TITLE
Why onSucess, onError, onSettled will be removed in next major version?

### DISCUSSION THREAD

**Discussion by SnowingFox** [TanStack/query#6139] [Q&A]
![image](https://github.com/TanStack/query/assets/26035718/a6392fc0-1039-4ccf-bfd2-7374124a59e3)

In my company project, we are using onSuccess, onError, onSettled in a lots of scenes, i know we can get the error message via `query.error` like this
```ts
const { error } = useQuery()

if (error) { ... }
```

But in my company's projects, they are using onSuccess, onError  to do something else when request is over, if you removed these options, we need to refactor the code or don't upgrade to the latest version, i think not only i will met this problem, but others, so i think this features can be saved

**Comment by TkDodo** [comment]
what's the other stuff you're doing?

please read:
- https://github.com/TanStack/query/discussions/5279
